File: backend/ResourcesAPI/Models/Data/Resource.cs

using System.ComponentModel.DataAnnotations;

namespace ResourcesAPI.Models.Data
{
    public class Resource
    {
        public Guid Id { get; set; }

        [Required, StringLength(254)]
        public string Name { get; set; } = null!;

        [DataType(DataType.Date)]
        public DateTime DateOfPublication { get; set; }

        [StringLength(254)]
        public string? Author { get; set; }

        [MaxLength(10)]
        public List<string> Tags { get; set; } = [];

        [StringLength(50)]
        public string Type { get; set; } = null!;

        [StringLength(500)]
        public string Description { get; set; } = null!;

        public int RentedItems { get; set; } = 0;   
        public int TotalItems { get; set; }=0;
    }
}

File: frontend/bookstore/src/app/resource/resource.ts
export class Resource { 
    id: string;
    name: string;
    dateOfPublication: Date;
    author: string;
    tags: string[];
    type: string;
    description: string;

    constructor(){
        this.id="";
        this.name="";
        this.author="";
        this.tags=[];
        this.type="";
        this.description="";
        this.dateOfPublication=new Date();
    }
}